### PR TITLE
Upgrade Celery to version 4.0.2

### PR DESCRIPTION
**Pull request by comzeradd**
This will need rebase after #1802 is merged

### CODE REVIEW

File: mozillians/celery.py [hunk 1]
```diff
@@ -9,7 +9,21 @@
...
+
+@app.on_after_configure.connect
+def setup_periodic_tasks(sender, **kwargs):
+    from mozillians.groups.tasks import (invalidate_group_membership,
+                                         notify_membership_renewal)
+
+    sender.add_periodic_task(RUN_DAILY, invalidate_group_membership.s(),
+                             name='invalidate-group-membership')
+
+    sender.add_periodic_task(RUN_DAILY, notify_membership_renewal.s(),
```
Comment (akatsoulas) [inline mozillians/celery.py]: These two are not the only periodic tasks that we have in the project.

**Comment by akatsoulas** [comment]
FYI, travis doesn't look very happy.

**Comment by comzeradd** [comment]
PR updated. I rebased to get the new periodic tasks we had in the meantime. I also included the `djcelery` removal parts to make Travis happy. Once we merge #1802 we'd need to rebase.

File: mozillians/settings/base_env_vars.py [hunk 2]
```diff
@@ -484,25 +483,12 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
...
-CELERY_TASK_RESULT_EXPIRES = config('CELERY_TASK_RESULT_EXPIRES', default=3600, cast=int)
+CELERY_ENABLE_UTC = config('CELERY_ENABLE_UTC', default=True, cast=bool)
 CELERY_SEND_TASK_ERROR_EMAILS = config('CELERY_SEND_TASK_ERROR_EMAILS', default=True, cast=bool)
+CELERY_RESULT_BACKEND = config('CELERY_RESULT_BACKEND', default='redis')
 REDIS_CONNECT_RETRY = config('REDIS_CONNECT_RETRY',
                              default=CELERY_RESULT_BACKEND == 'redis', cast=bool)
-# Time in seconds before celery.exceptions.SoftTimeLimitExceeded is raised.
-# The task can catch that and recover but should exit ASAP.
-CELERYD_TASK_SOFT_TIME_LIMIT = config('CELERYD_TASK_SOFT_TIME_LIMIT', default=150, cast=int)
+CELERY_BROKER_URL = config('CELERY_BROKER_URL', default='amqp://guest:guest@broker:5672//')
```
Comment (johngian) [inline mozillians/settings/base_env_vars.py]: This should be ``BROKER_URL`` http://docs.celeryproject.org/en/latest/userguide/configuration.html#std:setting-broker_url

http://docs.celeryproject.org/en/latest/userguide/configuration.html#new-lowercase-settings

File: mozillians/settings/base_env_vars.py [hunk 3]
```diff
@@ -484,25 +483,12 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
...
-CELERYBEAT_SCHEDULER = config('CELERYBEAT_SCHEDULER',
-                              default='djcelery.schedulers.DatabaseScheduler')
-
-# True says to simulate background tasks without actually using celeryd.
-# Good for local development in case celeryd is not running.
-CELERY_ALWAYS_EAGER = config('CELERY_ALWAYS_EAGER', default='True', cast=bool)
-BROKER_CONNECTION_TIMEOUT = config('BROKER_CONNECTION_TIMEOUT', default=0.1, cast=float)
-CELERY_RESULT_BACKEND = config('CELERY_RESULT_BACKEND', default='amqp')
-BROKER_URL = config('BROKER_URL', default='amqp://guest:guest@broker:5672//')
-CELERY_ACCEPT_CONTENT = config('CELERY_ACCEPT_CONTENT', default='pickle', cast=Csv())
```
Comment (johngian) [inline mozillians/settings/base_env_vars.py]: Let's keep this configurable. In reps we had some encoding issues and needed to fallback to pickle.

File: mozillians/settings/base_env_vars.py [hunk 4]
```diff
@@ -487,20 +487,16 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
...
-CELERY_TASK_RESULT_EXPIRES = config('CELERY_TASK_RESULT_EXPIRES', default=3600, cast=int)
+CELERY_ENABLE_UTC = config('CELERY_ENABLE_UTC', default=True, cast=bool)
+CELERY_TIMEZONE = config('CELERY_TIMEZONE', default='UTC')
+CELERY_TASK_RESULTS_EXPIRES = config('CELERY_TASK_RESULTS_EXPIRES', default=3600, cast=int)
+CELERY_ACCEPT_CONTENT = ['json']
 CELERY_SEND_TASK_ERROR_EMAILS = config('CELERY_SEND_TASK_ERROR_EMAILS', default=True, cast=bool)
+CELERY_RESULT_BACKEND = config('CELERY_RESULT_BACKEND', default='redis')
+CELERY_TASK_ALWAYS_EAGER = config('CELERY_TASK_ALWAYS_EAGER', default=False, cast=bool)
```
Comment (akatsoulas) [inline mozillians/settings/base_env_vars.py]: Actually this is not true. The default is False but we need this setting so we can override it in the env-dist vars for local development. Let's revert this

**Comment by akatsoulas** [comment]
This PR should be good to go with the last comments.

**Comment by comzeradd** [comment]
PR updated, with the two changes on `base_env_vars.py`.

**Comment by comzeradd** [comment]
PR rebased for the recent settings changes.

**Comment by comzeradd** [comment]
PR rebased after #1802 merge

**Comment by johngian** [comment]
Closing this one after #1994

**Pull request closed without merging**